sql/expression/function/reverse_repeat_replace.go

```diff
@@ -0,0 +1,253 @@
+package function
+
+import (
+	"fmt"
+	"strings"
+
+	"gopkg.in/src-d/go-mysql-server.v0/sql/expression"
+	"gopkg.in/src-d/go-mysql-server.v0/sql"
+	"gopkg.in/src-d/go-errors.v1"
+)
+
+// Reverse is a function that returns the reverse of the text provided.
+type Reverse struct {
+	expression.UnaryExpression
+}
+
+// NewLower creates a new Lower expression.
+func NewReverse(e sql.Expression) sql.Expression {
+	return &Reverse{expression.UnaryExpression{Child: e}}
+}
+
+// Eval implements the Expression interface.
+func (r *Reverse) Eval(
+	ctx *sql.Context,
+	row sql.Row,
+) (interface{}, error) {
+	v, err := r.Child.Eval(ctx, row)
+	if err != nil {
```

Comment:
This is not wrong but you can deal both cases of `nil` and error with one statement:

```go
if v == nil || err != nil {
      return nil, err
}
```